gitlab/resource_gitlab_deploy_key.go

```diff
@@ -98,8 +103,12 @@ func resourceGitlabDeployKeyDelete(d *schema.ResourceData, meta interface{}) err
 	}
 	log.Printf("[DEBUG] Delete gitlab deploy key %s", d.Id())
 
-	_, err = client.DeployKeys.DeleteDeployKey(project, deployKeyID)
+	response, err := client.DeployKeys.DeleteDeployKey(project, deployKeyID)
 
+	// HTTP 204 is success with no body
+	if response != nil && response.StatusCode == 204 {
```

Comment:
Why this change? The GitLab client does not error on 204.

Reply:
True! I'll revert that (cannot remember why sorry, it's been so long)